src/install/install.zig

```diff
@@ -252,9 +252,17 @@ const NetworkTask = struct {
         warn_on_error: bool,
     ) !void {
         this.url_buf = blk: {
+            var encoded_name = name;
+            // Not all registries support scoped package names when fetching the manifest.
+            // registry.npmjs.org supports both "@storybook%2Faddons" and "@storybook/addons"
+            // Other registries like AWS codeartifact only support the former.
+            // The NPM client reguests the the manifest with the encoded name.
+            if (strings.indexOfChar(encoded_name, '/')) |i| {
+                encoded_name = try strings.cat(allocator, try strings.cat(allocator, encoded_name[0..i], "%2F"), encoded_name[i + 1 ..]);
```

Comment:
Thank you for this

This can be done in one pass, without allocating (and potentially leaking) two strings. We can use a stack fallback allocator since typical usage here will be a small string. This bug is important to fix so I'll go ahead and make those changes.